File: Selenium/Activities/activity17.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.util.List;

public class activity17 {
    public static void main(String args[])
    {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver= new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/selects");
        System.out.println("Page title is: "+driver.getTitle());
        WebElement dropdown = driver.findElement(By.id("single-select"));
        Select select= new Select(dropdown);
        select.selectByVisibleText("Option 1");
        System.out.println("second option is:"+select.getFirstSelectedOption().getText());
        select.selectByIndex(2);
        System.out.println("Third option is: "+select.getFirstSelectedOption().getText());
        select.selectByValue("4");
        System.out.println("Fourth option is: "+select.getFirstSelectedOption().getText());
        List<WebElement> options = select.getOptions();
        for(WebElement option:options)
        {
            System.out.println(option.getText());
        }
        driver.close();

    }
}

File: Selenium/Activities/activity17.py
from selenium.webdriver.common.by import By
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
driver = webdriver.Firefox(service=FirefoxService(GeckoDriverManager().install()))
wait = WebDriverWait(driver, 10)
driver.get("https://v1.training-support.net/selenium/selects")
print("Page title is: ", driver.title)
single_select = driver.find_element(By.ID, "single-select")
dropdown = Select(single_select)

dropdown.select_by_visible_text("Option 2")
print("Selected option: ", dropdown.first_selected_option.text)

dropdown.select_by_index(3)
print("Selected option: ", dropdown.first_selected_option.text)

dropdown.select_by_value("4")
print("Selected option: ", dropdown.first_selected_option.text)

select_options = dropdown.options
for option in select_options:
    print("Option: ", option.text)
driver.close()
